File: packages/platos-client/src/apis/turns.ts
/** Canonical Platos Turn API. */

import type { PlatosClient } from "../client.js";
import type { PlatosScope } from "../types.js";

export interface PlatosTurn {
  id: string;
  agentId: string;
  threadId: string;
  status: string;
  createdAt: string;
  completedAt?: string | null;
  durationMs?: number;
  costCents?: number;
  [extra: string]: unknown;
}

export class TurnsApi {
  constructor(private readonly client: PlatosClient) {}

  async list(
    options: { agentId?: string; threadId?: string; status?: string; limit?: number } = {},
    scope?: PlatosScope,
  ): Promise<PlatosTurn[]> {
    const qs = new URLSearchParams();
    for (const [key, value] of Object.entries(options)) {
      if (value !== undefined && value !== null) qs.set(key, String(value));
    }
    const suffix = qs.toString() ? `?${qs.toString()}` : "";
    const res = await this.client._fetch<{ turns: PlatosTurn[] }>(
      `/api/v1/agent/turns${suffix}`,
      { method: "GET" },
      scope,
    );
    return res?.turns ?? [];
  }

  async get(turnId: string, scope?: PlatosScope): Promise<PlatosTurn> {
    return this.client._fetch<PlatosTurn>(
      `/api/v1/agent/turns/${encodeURIComponent(turnId)}`,
      { method: "GET" },
      scope,
    );
  }
}

File: packages/platos-client-py/platos_client/apis/turns.py
"""Canonical Platos Turn API."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any
from urllib.parse import quote, urlencode

if TYPE_CHECKING:
    from platos_client.client import PlatosClient, PlatosScope


class TurnsApi:
    def __init__(self, client: "PlatosClient") -> None:
        self._client = client

    async def list(
        self,
        *,
        agent_id: str | None = None,
        thread_id: str | None = None,
        status: str | None = None,
        limit: int | None = None,
        scope: "PlatosScope | None" = None,
    ) -> list[dict[str, Any]]:
        query = {key: value for key, value in {
            "agentId": agent_id,
            "threadId": thread_id,
            "status": status,
            "limit": limit,
        }.items() if value is not None}
        suffix = f"?{urlencode(query)}" if query else ""
        response = await self._client._request(
            "GET", f"/api/v1/agent/turns{suffix}", scope=scope
        )
        return list((response or {}).get("turns", []))

    async def get(
        self, turn_id: str, scope: "PlatosScope | None" = None
    ) -> dict[str, Any]:
        return await self._client._request(
            "GET", f"/api/v1/agent/turns/{quote(turn_id, safe='')}", scope=scope
        )
